server/src/main/java/org/opensearch/action/ActionModule.java

```diff
@@ -649,8 +649,10 @@ public <Request extends ActionRequest, Response extends ActionResponse> void reg
         actions.register(ClusterAllocationExplainAction.INSTANCE, TransportClusterAllocationExplainAction.class);
         actions.register(ClusterStatsAction.INSTANCE, TransportClusterStatsAction.class);
         actions.register(ClusterStateAction.INSTANCE, TransportClusterStateAction.class);
-        actions.register(GetTermVersionAction.INSTANCE, TransportGetTermVersionAction.class);
-        actions.register(ClusterHealthAction.INSTANCE, TransportClusterHealthAction.class);
+        if (FeatureFlags.isEnabled(FeatureFlags.CLUSTERLESS_FLAG) == false) {
```

Comment:
I don't have any immediate suggestions for a different way to do this, but the approach of adding conditional logic seems like it might become unmanageable if we start conditionally adding other actions. What do you think?

Also, if/when this feature graduates to non-experimental, would this condition be replaced by some sort of cluster setting?

Reply:
But in the long run, I think the (one?) remaining check in `Node` will look at a static cluster setting.